crates/bevy_winit/src/winit_windows.rs

```diff
@@ -281,52 +283,54 @@ pub fn get_fitting_videomode(
     width: u32,
     height: u32,
 ) -> winit::monitor::VideoMode {
-    let mut modes = monitor.video_modes().collect::<Vec<_>>();
-
     fn abs_diff(a: u32, b: u32) -> u32 {
         if a > b {
             return a - b;
         }
         b - a
     }
 
-    modes.sort_by(|a, b| {
-        use std::cmp::Ordering::*;
-        match abs_diff(a.size().width, width).cmp(&abs_diff(b.size().width, width)) {
-            Equal => {
-                match abs_diff(a.size().height, height).cmp(&abs_diff(b.size().height, height)) {
-                    Equal => b
-                        .refresh_rate_millihertz()
-                        .cmp(&a.refresh_rate_millihertz()),
-                    default => default,
+    monitor
+        .video_modes()
+        .min_by(|a, b| {
+            use std::cmp::Ordering::*;
+            match abs_diff(a.size().width, width).cmp(&abs_diff(b.size().width, width)) {
+                Equal => {
+                    match abs_diff(a.size().height, height).cmp(&abs_diff(b.size().height, height))
+                    {
+                        Equal => b
+                            .refresh_rate_millihertz()
+                            .cmp(&a.refresh_rate_millihertz()),
+                        default => default,
+                    }
                 }
+                default => default,
             }
```

Comment:
Since you're at it, this `match` could be replaced with [`Ordering::then_with`](https://doc.rust-lang.org/stable/std/cmp/enum.Ordering.html#method.then_with).

Another alternative would be using `min_by_key(|a| (abs_diff(a.size().width, width), abs_diff(a.size().height, height), Reverse(a.refresh_rate_millihertz())))`